force_wfmanager/left_side_pane/workflow_tree.py

```diff
@@ -317,21 +367,203 @@ def new_data_source(self, ui_info, object):
+        current_modelview_type = current_modelview.__class__.__name__
+
+        if current_modelview_type in mappings:
+            # Call self.verify_tree for any child modelviews, retrieving their
+            # error messages.
+            for child_modelview_list_name in mappings[current_modelview_type]:
+
+                child_modelview_list = getattr(current_modelview,
+                                               child_modelview_list_name)
+
+                for child_modelview in child_modelview_list:
+                    child_modelview_errors = self.verify_tree(mappings, errors,
+                                                              child_modelview)
+                    current_modelview.error_message += child_modelview_errors
+                    if current_modelview.error_message != '':
+                        current_modelview.valid = False
+
+        # Check current_modelview for errors
+        for error_pair in errors:
+            if current_modelview.model == error_pair.subject:
+                error_string = error_pair.error
+                current_modelview.error_message += (error_string + '\n')
+
+        if current_modelview.error_message != '':
+            current_modelview.valid = False
+            # Combine errors which refer to similar problems
+            current_modelview.error_message = collate_errors(current_modelview.
+                                                             error_message)
+        else:
+            current_modelview.valid = True
+
+        return current_modelview.error_message
+
     def modelview_editable(self, modelview):
         """Checks if the model associated to a ModelView instance
         has a non-empty, editable view """
         return model_info(modelview.model) != []
+
+    @on_trait_change("selected_mv,selected_mv.error_message,selected_mv.label")
+    def update_selected_error(self):
+
+        mv_label = self.selected_mv.label
+
+        if self.selected_mv is None:
+            self.selected_error = HTML_ERROR_TEMPLATE.format(
+                "No errors for {}.".format(mv_label), "")
+        elif self.selected_mv.error_message == '':
+            self.selected_error = HTML_ERROR_TEMPLATE.format(
+                "No errors for {}.".format(mv_label), "")
+        else:
+            error_list = self.selected_mv.error_message.split('\n')
+            body_strings = ''.join([HTML_SINGLE_ERROR.format(error)
+                                    for error in error_list])
+            self.selected_error = HTML_ERROR_TEMPLATE.format(
+                "Errors for {}:".format(mv_label), body_strings)
+
+
+def collate_errors(error_message):
+    """Group together similar error messages. For example, if output
+    parameters 1,2 and 3 have undefined names display 'Output parameters 1-3
+    have undefined names', rather than 3 separate error messages."""
+
+    #: List of error messages
+    error_list = error_message.split('\n')
```

Comment:
You should pass the list of error messages instead of splitting the already formed string. Compose the final string only at the very end, before displaying.